apps/src/code-studio/pd/teacher_application/teacher_application.jsx

```diff
@@ -0,0 +1,374 @@
+/* global window */
+
+/**
+ * Teacher Application questionaire
+ */
+
+import React from 'react';
+import _ from 'lodash';
+import {Radio, Checkbox, FormControl, FormGroup, ControlLabel} from 'react-bootstrap';
+
+function FieldGroup({ id, label, ...props }) {
+  return (
+    <FormGroup controlId={id}>
+      <ControlLabel>{label}</ControlLabel>
+      <FormControl {...props} />
+    </FormGroup>
+  );
+}
+
+const grades = ['Kindergarten'].concat(_.range(1,13));
+const subjects = ['Computer Science', 'Computer Literacy', 'Math', 'Science', 'History', 'English', 'Music', 'Art', 'Multimedia', 'Foreign Language'];
+const yesNoResposes = ['Yes', 'No'];
+
+const TeacherApplication = React.createClass({
+
+  getInitialState() {
+    return {
+      selectedCourse: ''
+    };
+  },
+
+  buttonList(type, label, groupName, answers, allowOther = false) {
```

Comment:
nit: I prefer prefixing render methods with `render` for clarity (and so the method has a verb), i.e. `renderButtonList`